site/src/pages/WorkspaceSettingsPage/WorkspaceSettingsForm.tsx

```diff
@@ -54,16 +58,20 @@ export const WorkspaceSettingsForm: FC<{
     <HorizontalForm onSubmit={form.handleSubmit} data-testid="form">
       <FormSection
         title="Workspace Name"
-        description="Update the name of your workspace."
+        description="The name of your workspace."
       >
         <FormFields>
           <TextField
             {...getFieldHelpers("name")}
-            disabled={form.isSubmitting}
+            disabled={!workspace.allow_renames || form.isSubmitting}
             onChange={onChangeTrimmed(form)}
             autoFocus
             fullWidth
             label="Name"
+            helperText={
+              !workspace.allow_renames &&
+              "Renaming your workspace can be destructive and has not been enabled for this deployment."
+            }
```

Comment:
We have an alert below that relies on slightly different logic but has similar messaging. It would be nice to consolidate and just use helper text IMO. You could even conditionally change the color of the text so that it assumes a warning hue if `workspace.allow_renames` is true. 
![Screenshot 2023-12-15 at 11 17 40 AM](https://github.com/coder/coder/assets/19142439/d6e64fe7-e1c3-4cc2-b7ab-3831cd3e0271)

The class to target would be `.MuiFormHelperText-root`